src/inputs/Checkbox/Checkbox.tsx

```diff
@@ -1,23 +1,25 @@
 'use client'
+
 import React from 'react'
 import styles from './Checkbox.module.scss'
 import { IoIosCheckmark } from 'react-icons/io'
 
 interface CheckboxInputProps {
   label: string
   checked: boolean
-  onChange: (checked: boolean) => void
+  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
+  name: string
 }
 
 const CheckboxInput: React.FC<CheckboxInputProps> = ({
   label,
   checked,
-  onChange,
+  onChange, 
+  name,
 }) => {
-  const handleCheckboxChange = () => {
-    onChange(!checked)
+  const handleCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    onChange(event)
   }
-
   return (
     <label className={styles.checkboxLabel}>
```

Comment:
**issue:** label is missing `htmlFor`, it can be the issue why validation works incorrectly